src/charm.py

```diff
@@ -23,7 +23,7 @@
 logger = logging.getLogger(__name__)
 
 
-@trace_charm(tempo_endpoint="_tempo_otlp_grpc_endpoint", service_name="TempoCharm")
+@trace_charm(tempo_endpoint="_tempo_otlp_grpc_endpoint")
```

Comment:
Is there anything other than 'charm'? I.e. should it be shortened to just `@trace(...)`?

Reply:
didn't see this one until now, but yeah, there's `trace_function, trace_type`.
Idea is:


```python
@trace_function
def my_custom_fn()...

@trace_type
class MyObj...

@trace_charm(...)
 class MyCharm...
     def foo(self):
          my_custom_fn()  # we'll open a span corresponding with `my_custom_fn` being called
          obj = MyObj()  # each method call on this type will open a span
          obj.bar()
```

In fact, `trace_charm` is a specialization of `trace_type`.
I thought about having a `@trace` that automagically dispatches to one of the three based on the type (and subclass) of the first argument, but it felt out of scope for now